corehq/apps/locations/forms.py

```diff
@@ -305,17 +273,17 @@ def clean_coordinates(self):
 
         return [lat, lon]
 
-    def save(self, instance=None, commit=True):
+    def save(self, metadata=None):
         if self.errors:
             raise ValueError('form does not validate')
 
-        location = instance or self.location
+        location = self.location
         is_new = location.location_id is None
 
         location.name = self.cleaned_data['name']
         location.site_code = self.cleaned_data['site_code']
         location.location_type = self.cleaned_data['location_type_object']
-        location.metadata = self.custom_data.get_data_to_save()
+        location.metadata = metadata or {}
```

Comment:
So if `metadata` isn't passed in (as [here](https://github.com/dimagi/commcare-hq/pull/15774/files#diff-1d8a6e2b00dbb7850a4c1bfde042d6e8R523)), `location.metadata` gets cleared out? Making metadata a required parameter would make this clearer.